catalog/catalog.yaml

```diff
@@ -580,6 +582,54 @@ components:
         passThrough:
           type: boolean
 
+    BatchWriteSupport:
+      required:
+        - delete
+        - create
+        - update
+        - upsert
+      properties:
+        delete:
+          $ref: '#/components/schemas/BatchWriteSupportConfig'
+        create:
+          $ref: '#/components/schemas/BatchWriteSupportConfig'
+        update:
+          $ref: '#/components/schemas/BatchWriteSupportConfig'
+        upsert:
+          $ref: '#/components/schemas/BatchWriteSupportConfig'
+
+    BatchWriteSupportConfig:
+      required:
+        - supported
+      properties:
+        supported:
+          description: Whether this type of batch write operation is supported
+          type: boolean
+        defaultRecordLimit:
+          description: The default number of records supported in a batch
+          type: integer
+        objectRecordLimits:
+          description: Any object-level overrides for number of records supported in a batch
```

Comment:
this should be `*map[string]int` (converted to openapi format), where the key of the map is the object name and the value of the map is the record limit